File: Lecture10_Threads/AfterWorkshop_Java/app/src/main/java/com/lavima/afterworkshop_java/MainActivity.java
package com.lavima.afterworkshop_java;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.util.Log;
import android.widget.ImageView;
import android.widget.TextView;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class MainActivity extends AppCompatActivity {

    TextView titleText;
    ImageView posterImage;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        titleText = findViewById(R.id.titleText);
        posterImage = findViewById(R.id.posterImage);

        //GetMovieAsync getMovieAsync = new GetMovieAsync(titleText, posterImage);
        //getMovieAsync.execute("https://www.omdbapi.com/?i=tt3896198&apikey=2f6990a0");

        Thread getMoviewThred = new Thread(new Runnable() {
            @Override
            public void run() {

                try {
                    URL url = new URL("https://www.omdbapi.com/?i=tt3896198&apikey=2f6990a0");

                    HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();

                    urlConnection.setRequestMethod("GET");
                    urlConnection.connect();

                    BufferedReader br = new BufferedReader(new InputStreamReader(url.openStream()));

                    StringBuilder sb = new StringBuilder();
                    String line;
                    while ((line = br.readLine()) != null) {
                        sb.append(line).append("\n");
                    }
                    br.close();
                    urlConnection.disconnect();

                    String jsonString = sb.toString();
                    JSONObject jsonObject = new JSONObject(jsonString);

                    InputStream in = new java.net.URL(jsonObject.getString("Poster")).openStream();
                    Bitmap poster = BitmapFactory.decodeStream(in);

                    updateUI(jsonObject.getString("Title"), poster);
                }
                catch (JSONException jsone) {
                    Log.e("Worker.getJSON", jsone.getMessage());
                }
                catch (IOException ioe) {
                    Log.e("Worker.getJSON", ioe.getMessage());
                }
            }
        });
        getMoviewThred.start();
    }

    private void updateUI(final String title, final Bitmap poster) {
        runOnUiThread(new Runnable() {
            @Override
            public void run() {
                titleText.setText(title);
                posterImage.setImageBitmap(poster);
            }
        });
    }

}

File: Lecture07_Navigation_UI/Before_Kotlin_01_Binding/app/src/main/java/no/hiof/larseknu/playingwithmaterialdesign/MainActivity.kt
package no.hiof.larseknu.playingwithmaterialdesign

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    private lateinit var countTextView : TextView
    private lateinit var countButton : Button

    private var counter = 0


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.i("MainActivity", "onCreate Called")
        setContentView(R.layout.activity_main)

        countTextView = findViewById(R.id.countTextView)
        countButton = findViewById(R.id.countButton)

        if (savedInstanceState != null) {
            counter = savedInstanceState.getInt("COUNTER_ID")

            countTextView.text = counter.toString()
        }

        countButton.setOnClickListener{ view ->
            counter++

            countTextView.setText(counter.toString())
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)

        outState.putInt("COUNTER_ID", counter)
    }

    override fun onStart() {
        super.onStart()
        Log.i("MainActivity", "onStart called")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.i("MainActivity", "onDestroy Called")
    }
}
